src/finch/compiled.py

```diff
@@ -0,0 +1,35 @@
+from functools import wraps
+
+from .julia import jl
+from .tensor import Tensor
+
+
+def compiled(func):
+    @wraps(func)
+    def wrapper_func(*args, **kwargs):
+        new_args = []
+        for arg in args:
+            if isinstance(arg, Tensor) and not jl.isa(arg._obj, jl.Finch.LazyTensor):
```

Comment:
@willow-ahrens 
> Let's move this wrapper_func functionality out so that we have a function lazy that wraps tensors in LazyTensor.

Reply:
I think I misunderstood this, yes this is okay!